src/explore-education-statistics-admin/src/components/editable/EditableKeyStat.tsx

```diff
@@ -1,205 +1,81 @@
-
-  const formId = `editableKeyStatForm-${dataBlockId}`;
-
-  const renderInner = () => {
-    if (error || !keyStat) {
-      return (
-        <>
-          <WarningMessage>Could not load key stat</WarningMessage>
-
-          <ButtonGroup>
-            {onRemove && (
-              <Button
-                disabled={removing}
-                variant="secondary"
-                onClick={() => {
-                  toggleRemoving.on();
-                  onRemove();
-                }}
-              >
-                Remove
-              </Button>
-            )}
-          </ButtonGroup>
-        </>
-      );
-    }
-
-    if (showForm) {
-      return (
-        <Formik<KeyStatsFormValues>
-          initialValues={{
-            dataSummary: summary?.dataSummary?.[0] ?? '',
-            dataDefinitionTitle: summary?.dataDefinitionTitle?.[0] ?? 'Help',
-            dataDefinition: summary?.dataDefinition?.[0]
-              ? toHtml(summary?.dataDefinition?.[0])
-              : '',
-          }}
-          onSubmit={values => {
-            onSubmit({
-              ...values,
-              dataDefinitionTitle: values.dataDefinitionTitle.trim(),
-              dataDefinition: toMarkdown(values.dataDefinition),
-            });
-            toggleShowForm.off();
-          }}
-        >
-          {form => (
-            <Form id={formId}>
-              <h3 className="govuk-heading-s">{name}</h3>
-
-              <KeyStatTile
-                title={keyStat.title}
-                titleTag="h4"
-                testId={testId}
-                value={keyStat.value}
-                isReordering={isReordering}
-              >
-                <FormFieldTextInput<KeyStatsFormValues>
-                  name="dataSummary"
-                  label={<span className={styles.trendText}>Trend</span>}
-                />
-              </KeyStatTile>
-
-              <FormFieldTextInput<KeyStatsFormValues>
-                formGroupClass="govuk-!-margin-top-2"
-                name="dataDefinitionTitle"
-                label="Guidance title"
-              />
-
-              <FormFieldEditor<KeyStatsFormValues>
-                name="dataDefinition"
-                toolbarConfig={toolbarConfigs.simple}
-                label="Guidance text"
-              />
-
-              <ButtonGroup>
-                <Button
-                  disabled={!form.isValid}
-                  type="submit"
-                  className="govuk-!-margin-right-2"
-                >
-                  Save
-                </Button>
-                <Button variant="secondary" onClick={toggleShowForm.off}>
-                  Cancel
-                </Button>
-              </ButtonGroup>
-            </Form>
-          )}
-        </Formik>
-      );
-    }
-
-    const dataDefinitionTitle = summary?.dataDefinitionTitle[0] || 'Help';
-
+  const { updateKeyStatisticDataBlock } = useReleaseContentActions();
+  if ((keyStat as KeyStatisticDataBlock).dataBlockId) {
```

Comment:
Don't think this is quite the right way to solve this type problem. We should probably consider writing a type guard to test whether the keyStat is of type `KeyStatisticsDataBlock` rather than casting. There's an example on how to do this for multiple items in `src/explore-education-statistics-common/src/utils/type-guards` (think we'd want to write a new function to test for one item).

Reply:
We already use Type guards across the code base with the `is` operator which work fine in a production setting. https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates